src/lib/RequestParser.js

```diff
@@ -0,0 +1,194 @@
+        if (!keyInfo) {
+            throw new Errors.KeyNotFoundError();
+        }
+        return keyInfo;
+    }
+
+    /**
+     * @param {any} indicesArray
+     * @returns {string[]}
+     */
+    parseIndicesArray(indicesArray) {
+        if (!indicesArray || !Array.isArray(indicesArray)) {
+            throw new Errors.InvalidRequestError('indicesToDerive must be an array');
+        }
+        if (indicesArray.length === 0) {
+            throw new Errors.InvalidRequestError('indicesToDerive must not be empty');
+        }
+        indicesArray.forEach(/** @param {any} index */index => {
+            if (typeof index !== 'string') {
+                throw new Errors.InvalidRequestError('indicesToDerive must consist of strings');
+            }
+            if (!Nimiq.ExtendedPrivateKey.isValidPath(`m/${index}`)) {
+                throw new Errors.InvalidRequestError(
+                    'indicesToDerive strings must start with a number and end with a \'',
+                );
+            }
+        });
+        return indicesArray;
+    }
+
+    /**
+     * @param {any} object
+     * @returns {Nimiq.ExtendedTransaction}
+     */
+    parseTransaction(object) {
+        const accountTypes = new Set([Nimiq.Account.Type.BASIC, Nimiq.Account.Type.VESTING, Nimiq.Account.Type.HTLC]);
+        if (!object || typeof object !== 'object' || object === null) {
+            throw new Errors.InvalidRequestError('Request must be an object');
+        }
+
+        const sender = this.parseAddress(object.sender);
+        const senderType = object.senderType || Nimiq.Account.Type.BASIC;
+        if (!accountTypes.has(senderType)) {
+            throw new Errors.InvalidRequestError('Invalid sender type');
+        }
+
+        const recipient = this.parseAddress(object.recipient);
+        const recipientType = object.recipientType || Nimiq.Account.Type.BASIC;
+        if (!accountTypes.has(recipientType)) {
+            throw new Errors.InvalidRequestError('Invalid sender type');
+        }
+
+        if (sender.equals(recipient)) {
+            throw new Errors.InvalidRequestError('Sender and recipient must not match');
+        }
+
+        const flags = object.flags || Nimiq.Transaction.Flag.NONE;
+        const data = object.data || new Uint8Array(0);
+        return new Nimiq.ExtendedTransaction(
+            sender,
+            senderType,
+            recipient,
+            recipientType,
+            object.value,
+            object.fee,
+            object.validityStartHeight,
+            flags,
+            data,
+        );
+    }
+
+    /**
+     * @param {any} address
+     * @returns {Nimiq.Address}
+     */
+    parseAddress(address) {
+        try {
+            const nqAddress = new Nimiq.Address(address);
+            return nqAddress;
+        } catch (error) {
+            throw new Errors.InvalidRequestError(`recipient must be a valid Nimiq Address (${error.message})`);
```

Comment:
You need to pass the parameter name into the function, to be able to differenciate between sender and recipient in the error message.